packages/date-picker/src/DatePicker/DatePicker.tsx

```diff
@@ -149,71 +144,66 @@ export const DatePicker: React.FunctionComponent<DatePickerProps> = ({
     setIsOpen(false)
   }
 
+  const handleOpenClose = () => {
+    setIsOpen(!isOpen)
+  }
+
+  const handleReturnFocus = () => {
+    if (wrapperRef && wrapperRef.current) {
+      const buttonElement = wrapperRef.current.getElementsByClassName(
+        datePickerStyles.button
+      )[0] as HTMLElement
+
+      buttonElement?.focus()
+    }
+  }
```

Comment:
Frustratingly (is that a word?) the `returnFocus` prop in `react-focus-lock` doesn't work out of the box all the time (and people have complained) so it accepts a function.

Also, I would've loved to use `buttonRef` here but as it's something a consumer can pass in, I've had to do the outer wrapper.